src/Models/SharedUtils/DataHistoryUtils.ts

```diff
@@ -176,3 +184,46 @@ export const getDefaultSeriesLabel = (
 ): string => {
     return `${twinId} (${propertyName})`;
 };
+
+/** send the KPI telemetry of captured data history explorer modal metrics  */
+export const sendDataHistoryExplorerSystemTelemetry = (
+    timeSeriesTwinList: Array<IDataHistoryTimeSeriesTwin>
+) => {
+    // capture the Data History Explorer Modal level metrics
+    const event =
+        TelemetryEvents.Tools.DataHistoryExplorer.SystemAction.ParseModal;
+    TelemetryService.sendEvent(
+        new TelemetryEvent({
+            name: event.eventName,
+            appRegion: AppRegion.DataHistoryExplorer,
+            componentName: ComponentName.DataHistoryExplorerModal,
+            customProperties: {
+                [event.properties.countSeries]: timeSeriesTwinList.length
+            },
+            triggerType: TelemetryTrigger.SystemAction
+        })
+    );
+};
+
+/** send the user actions telemetry  */
+export const sendDataHistoryExplorerUserTelemetry = (
```

Comment:
I like this pattern for sending events that have the same keys in custom properties.